driver/pelican/types/scheduleSet.go

```diff
@@ -0,0 +1,113 @@
+package types
+
+import (
+	"encoding/xml"
+	"fmt"
+
+	"github.com/parnurzeal/gorequest"
+	rrule "github.com/teambition/rrule-go"
+)
+
+func (pel *Pelican) SetSchedule(newSchedule *ThermostatSchedule) error {
+	// Retrieve thermostat's latest schedule for extra block deletion purposes
+	currentSchedule, currentErr := pel.GetSchedule()
+	if currentErr != nil {
+		return fmt.Errorf("Error retrieving thermostat %v's current schedule: %v", pel.Name, currentErr)
+	}
+
+	for _, day := range week {
+		if blocks, ok := newSchedule.DaySchedules[day]; ok {
+			// Delete Unnecessary Blocks + Error Checking
+			currentBlockCount := len(currentSchedule.DaySchedules[day])
+			requiredBlockCount := len(blocks)
+			for requiredBlockCount < currentBlockCount {
+				respDelete, _, errsDelete := gorequest.New().Get(pel.target).
+					Param("username", pel.username).
+					Param("password", pel.password).
+					Param("request", "set").
+					Param("object", "thermostatSchedule").
+					Param("selection", fmt.Sprintf("name:%s;dayOfWeek:%s;setTime:%v;", pel.Name, day, currentBlockCount)).
+					Param("value", "delete").
+					End()
+
+				if errsDelete != nil {
+					return fmt.Errorf("Error deleting thermostat schedule settings on day %v: %v\n", day, errsDelete)
+				}
+				defer respDelete.Body.Close()
+				var resultDelete apiResult
+				decDelete := xml.NewDecoder(respDelete.Body)
+				if errDecodeDelete := decDelete.Decode(&resultDelete); errDecodeDelete != nil {
+					return fmt.Errorf("Failed to decode thermostat schedule delete response XML: %v", errDecodeDelete)
+				}
+				if resultDelete.Success == 0 {
+					return fmt.Errorf("Error deleting thermostat schedule settings on day %v: %v\n", day, resultDelete.Message)
+				}
+				currentBlockCount -= 1
+			}
+
+			// Construct and Set New Schedule Settings by Block
+			for index, block := range blocks {
+				var value string = ""
+				value += fmt.Sprintf("coolSetting:%.0f;", block.CoolSetting)
+				value += fmt.Sprintf("heatSetting:%.0f", block.HeatSetting)
+				value += fmt.Sprintf("system:%s", block.System)
+
+				// Convert Time to Pelican's Timezone
+				time, timeErr := rrule.StrToRRule(block.Time)
```

Comment:
Let's use a different name for this variable to avoid confusion with the `time` package.

Reply:
Sorry about the late update. Thanks for the catch! I changed the name to timeRRule and timeRRuleErr instead.